.storybook/main.js

```diff
@@ -0,0 +1,20 @@
+module.exports = {
+  stories: ['../stories/**/*.stories.tsx'],
+  addons: ['@storybook/addon-knobs/register'],
+  webpackFinal: async config => {
+    config.module.rules.push({
+      test: /\.(ts|tsx)$/,
+      use: [
+        {
+          loader: require.resolve('ts-loader'),
+        },
+        // Optional
+        {
+          loader: require.resolve('react-docgen-typescript-loader'),
+        },
+      ],
+    });
+    config.resolve.extensions.push('.ts', '.tsx');
+    return config;
+  },
+};
```

Comment:
@demiankatz Here's the missing file, you should be able to run it locally now. Sorry about that, problems with the .gitignore for `main.js` files.

Reply:
Well, that's progress, but it's still not quite working for me. Now, if I run `grunt build` and `npm run storybook`, I see the full storybook interface with the two examples you provided and the appropriate knobs, but the UV itself doesn't load. The pane where it should be is white and empty. I'm seeing a 404 error in my console for the file `http://localhost:6006/uv/lib/uv-openseadragon-extension.en-GB.config.json`.

I haven't tried deleting everything and starting fresh with a clean `npm install`. If you think that's worth a shot, let me know and I'll do it... but since it's a time-consuming process, I first wanted to check and see if the missing JSON file was an obvious indication of a different problem.

Reply:
For what it's worth, on the working Netlify demo you shared, the config.json file is coming from the URL `https://uv-storybook.netlify.com/uv-assets/config/uv-openseadragon-extension.en-GB.config.json`. Not sure why the path is so significantly different in my local environment.

Reply:
Hmm, can you try: `grunt build --dist` and then `npm run storybook` ?

Reply:
Same result. I have a `./dist` directory which is being served up by Storybook through port 6006... but something is still trying to request `http://localhost:6006/uv/lib/uv-openseadragon-extension.en-GB.config.json` and there's no `./dist/uv` directory in my instance, so I get a 404.

Reply:
Hmm, not sure I know what's causing that. Any ideas @edsilv ?

Reply:
I'd definitely be interested to hear whether @edsilv has been able to get this running on his own system. If it works for him, then I'm happy to blame my setup... but if not, maybe there's still something not quite right in this PR relative to your local system.

Reply:
Yes, I was certainly having issues with the `.gitignore` picking up some of the configuration, could still be something I've missed!

Reply:
I just tried running it (`npm build --dist && npm run storybook`)  and am getting:

```
ERROR in ./src/Viewer.js
Module not found: Error: Can't resolve './extensions/uv-virtex-extension/Extension' in 'C:\Users\edsilv\github\universalviewer\universalviewer-webpack\src'
 @ ./src/Viewer.js 357:16-361:59
 @ ./src/index.js
 @ ./stories/uv.stories.tsx
 @ ./stories sync ^\.\/(?:(?:(?!\.)(?:(?:(?!(?:|[\\/])\.).)*?)[\\/])?(?!\.)(?=.)[^\\/]*?\.stories\.tsx[\\/]?)$
 @ ./.storybook/generated-entry.js
 @ multi ./node_modules/@storybook/core/dist/server/common/polyfills.js ./node_modules/@storybook/core/dist/server/preview/globals.js ./.storybook/generated-entry.js (webpack)-hot-middleware/client.js?reload=true&quiet=true

```

Reply:
it's weird. runs fine with `npm start`

Reply:
@edsilv, did you mean `grunt build --dist`? I tried running `npm build --dist` and just got an error message.

Reply:
`npm run build --dist`

Reply:
@edsilv, I don't get an error when I do that, but I don't think the `--dist` flag carries through when you wrap `npm` around the command.

Reply:
(I think `npm run build:dist` is the way to do it through `npm` as currently configured. But that works for me too. Perhaps the virtex error you're seeing has to do with an artifact in your local system? Did you try from a clean checkout?